test-requirements.txt

```diff
@@ -0,0 +1,2 @@
+pytest
```

Comment:
Should not we add `mypy` to `test-requirements.txt`, although I've run `pip install -r test-requirements.txt`, I was not able to run `mypy --ignore-missing-imports .`, in my opinion this could be confusing. @sfc-gh-zblackwood what do you think?

Reply:
Same comment is valid for ruff

Reply:
Yup